private/buf/buflsp/server.go

```diff
@@ -411,9 +411,9 @@ func (s *server) SemanticTokensFull(
 	defer progress.Done(ctx)
 
 	var (
-		encoded           []uint32
 		prevLine, prevCol uint32
 	)
+	encoded := []uint32{}
```

Comment:
Can you point to how this makes a difference? Downstream code should generally not care whether this is 0 or nil. If there's a specific reason, this should be documented here with a code comment so that we understand this choice in the future.

Reply:
Updated the PR, title, and description: I did some digging into the specification, and `textDocument/semanticTokens/full` expects `SemanticTokens | null` as the response, but `data` on `SemanticTokens` is not optional. So in this case, if we are not sending any `data`, we should instead be returning `nil` for semantic tokens. I added this explanation as in-line comments in the code change as well.